### PR TITLE
Initial commit of docker-cloud project

### PR DESCRIPTION
**Pull request by tgvashworth**
This project will set up using Docker and containers. It will jump directly to ECS and Fargate, which hide some details of cloud VMs such as EC2. We'll need to return to that topic later on to adhere to the teach objectives of the course. However, it's expedient to do this now so that other projects can rely on docker and cloud deployment knowledge.

Address #37 

-----
[View rendered README.md](https://github.com/CodeYourFuture/immersive-go-course/blob/readme-docker-cloud/README.md)
[View rendered docker-cloud/README.md](https://github.com/CodeYourFuture/immersive-go-course/blob/readme-docker-cloud/docker-cloud/README.md)

### CODE REVIEW

File: docker-cloud/README.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
+# Docker & Cloud Deployment
+
+In this project, you'll build a simple Go server application and Dockerise it to run within a container. You'll write tests that run against the container, and then build GitHub actions automate continuous testing and release of the application. You'll then run it in the cloud by pushing the container image to Amazon AWS Elastic Container Repository, and then launch it in Amazon AWS Elastic Container Service using the user interface.
```
Comment (illicitonion) [inline docker-cloud/README.md:3]:
```suggestion
In this project, you'll build a simple Go server application and Dockerise it to run within a container. You'll write tests that run against the container, and then build GitHub actions to automate continuous testing and release of the application. You'll then run it in the cloud by pushing the container image to Amazon AWS Elastic Container Repository, and then launch it in Amazon AWS Elastic Container Service using the user interface.
```

File: docker-cloud/README.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+> ⚠️ This project requires you to have access to an Amazon AWS account, with permissions to configure ECS, ECR, Fargate, and Elastic Load Balancing. Ask on CYF Slack for help with that.
+
+Learning objectives:
+
+- Det up Docker locally
```
Comment (illicitonion) [inline docker-cloud/README.md:9]:
```suggestion
- Set up Docker locally
```

File: docker-cloud/README.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+Following [docs here](https://github.com/ory/dockertest) and [example here](https://github.com/olliefr/docker-gs-ping), write some tests.
+
+Go dependencies are installed by `go get`.
+
+Make sure to `COPY go.sum ./` in your `Dockerfile`. [Read this guide to understand why](https://golangbyexample.com/go-mod-sum-module/).
+
+## CI/CD
+
+A set of good practices has exists for developing software for production use by thousands or millions of people. One example of this is Continuous Integration & Continuous Deployment, referred to as CI/CD. We're going to focus on the "CI" component of this, which means:
```
Comment (illicitonion) [inline docker-cloud/README.md:156]:
```suggestion
A set of good practices exists for developing software for production use by thousands or millions of people. One example of this is Continuous Integration & Continuous Deployment, referred to as CI/CD. We're going to focus on the "CI" component of this, which means:
```

**Review by illicitonion — APPROVED** [review]
LGTM - thanks!

**Comment by lauranooooo** [comment]
This looks generally great.
If you wanted to add a small 'extra reading' section Julia Evans has a couple of useful posts on containers.
https://jvns.ca/blog/2016/10/10/what-even-is-a-container/

File: docker-cloud/README.md [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+- Containers are your running application, based on an image
+
+### Cloud hosting in AWS
+
+We're going to host our application in the cloud, specifically in Amazon Web Services (AWS). AWS is a large suite of products for running technology in the cloud.
+
+The set of AWS products we're going to interact with directly are:
+
+- Elastic Container Repository (ECS): store images that can later be run as containers
+- Elastic Container Service (ECR): run containers, including all the infrastructure needed to make them accessible to the internet
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply docker-cloud/README.md:85]: Oop yes thanks

**Pull request merged**